src/components/organism/ToDoList/ToDoList.tsx

```diff
@@ -1,79 +1,152 @@
+
+const ToDoList = ({selectionMode,UpdateCallback,searchedTodos,addToDo,search}: ToDoListProps) => {
+    //global state>
+    const reduxToDos = useSelector(selectAlltodos)
     const dispatch = useDispatch()
     //<
-    //local states >
+    //compoment state>
     const [selectedToDos,setSelectedToDos] = useState<todoprops[]>([]);
     const [targetToDo,setTargetToDo] = useState<todoprops>({} as todoprops);
     const [deletedrawer,setDeletedrawer] = useState<boolean>(true);
+    let sortedToDos:todoprops[] = []
     //<
-    //sorts between complete and uncomplete >
-    const uncompletedtodos = todos.filter(currentTodo =>{return !currentTodo.complete})
-    const completedtodos = todos.filter(currentTodo =>{return currentTodo.complete})
-    //<
-    //Additional helper func>
-    const DrawerOnYes = ()=> {Toast({message:"To Do Deleted."}); dispatch(deletetodo(targetToDo))}
-    const SDrawerOnDeleteSelected = () =>{Toast({message:"To Dos Deleted."}); dispatch(deletetodos([...selectedToDos]))}
-    const SDrawerOnCompleteSelected = () =>{Toast({message:"To Dos Completed."});dispatch(completetodos([...selectedToDos]))}
-    
-    //<
-        const todoitem = (todo: todoprops) => {
-            //Additional helper func>
-            const todoitemOnSelect = () => { if(selectedToDos.includes(todo)){
-                setSelectedToDos(selectedToDos.filter(currentTodo =>{ return !(currentTodo.id===todo.id)}))
-                } else{ 
-                setSelectedToDos(prevselectedToDos => [...prevselectedToDos,todo])
-                }
+    //override/sort>
+        if(search){
+            if(searchedTodos===undefined){
+            }else {
+                if(searchedTodos.length > 0){
+                    sortedToDos = [...searchedTodos];
+                    }
             }
-            const todoitemOnTextClick = () => {setTargetToDo(todo);dispatch(completetodo(targetToDo))}
-            const todoitemOnDelete = () =>{ setTargetToDo(todo);setDeletedrawer(false)}
-            const todoitemOnUpdate = () =>{ Toast({message:"To Do Updated."}); UpdateCallback()}
+        }
+        else {
+            //sorts between complete and uncomplete >
+            const uncompletedtodos = reduxToDos.filter(currentTodo =>{return !currentTodo.complete})
+            const completedtodos = reduxToDos.filter(currentTodo =>{return currentTodo.complete})
+            sortedToDos = [...uncompletedtodos,...completedtodos]
             //<
-            return <ToDoItem 
-            onSelect={()=>todoitemOnSelect()} 
-            selected={selectedToDos.includes(todo)} 
-            selectionMode={selectionMode} 
-            key={todo.id} 
-            value={todo.todotext} 
-            completed={todo.complete}
-            onTextClick={()=>{todoitemOnTextClick()}}
-            onDelete={()=>{todoitemOnDelete()}}
-            onUpdate={()=>{todoitemOnUpdate()}}/>
         }
-
+    //<
+    //>
+    const isFilled = sortedToDos.length;
```

Comment:
Please add double exclamation point `!!sortedToDos.length` if you only need the boolean for `isFilled`.